evaluator.py

```diff
@@ -0,0 +1,243 @@
+        return recall_score(y_true=self.y, y_pred=self.preds)
+
+    def get_f1_score(self) -> float:
+        return f1_score(y_true=self.y, y_pred=self.preds)
+
+    def get_pr_auc(self) -> float:
+        precision, recall, _ = self._get_pr_curve_properties()
+        return auc(recall, precision)
+
+    def get_all_metrics(self, to_mlflow: bool = False) -> Metrics:
+        metrics = Metrics(
+            accuracy=self.get_accuracy(),
+            roc_auc=self.get_roc_auc(),
+            precision=self.get_precision(),
+            recall=self.get_recall(),
+            f1=self.get_f1_score(),
+            pr_auc=self.get_pr_auc(),
+        )
+
+        if to_mlflow:
+            mlflow.log_metrics(metrics)
+
+        return metrics
+
+    def plot_roc_curve(self, save_path: Path = None) -> Figure:
+        fpr, tpr, thresholds = roc_curve(y_true=self.y, y_score=self.proba)
+        auc_score = auc(fpr, tpr)
+
+        gmeans = np.sqrt(tpr * (1 - fpr))
+        ix = np.argmax(gmeans)
+
+        fig, ax = plt.subplots(1, 1)
+        fig.set_size_inches(15, 7)
+
+        ax.plot([0, 1], [0, 1], linestyle="--", label="Random model")
+        ax.plot(fpr, tpr, marker=".", label=f"Model - AUC: {auc_score:.3f}")
+        ax.scatter(
+            fpr[ix],
+            tpr[ix],
+            marker="o",
+            color="black",
+            label=f"Best Threshold={thresholds[ix]:.2f}, G-Mean={gmeans[ix]:.2f}",
+        )
+
+        ax.set_xlabel("False Positive Rate")
+        ax.set_ylabel("True Positive Rate")
+        ax.legend(loc="upper left")
+
+        if save_path:
+            fig.savefig(save_path / "roc_curve.png")
+
+        return fig
+
+    def plot_precision_recall_curve(self, save_path: Path = None) -> Figure:
+        precision, recall, thresholds = precision_recall_curve(
+            self.y, self.proba
+        )
+        auc_score = self.get_pr_auc()
+
+        fscore = (2 * precision * recall) / (precision + recall)
+        ix = np.argmax(fscore)
+
+        no_skill = len(self.y[self.y == 1]) / len(self.y)
+
+        fig, ax = plt.subplots(1, 1)
+        fig.set_size_inches(15, 7)
+
+        ax.plot([0, 1], [no_skill, no_skill], linestyle="--", label="No Skill")
+        ax.plot(
+            recall,
+            precision,
+            marker=".",
+            label=f"Model - AUC: {auc_score:.3f}",
+        )
+        ax.scatter(
+            recall[ix],
+            precision[ix],
+            marker="o",
+            color="black",
+            label=f"Best Threshold={thresholds[ix]:.2f}, F-Score={fscore[ix]:.2f}",
+        )
+
+        ax.set_xlabel("Recall")
+        ax.set_ylabel("Precision")
+        ax.legend(loc="upper left")
+
+        if save_path:
+            fig.savefig(save_path / "precision_recall_curve.png")
+
+        return fig
+
+    def plot_partial_dependency_plot(
+        self, save_path: Path = None, kind: str = "average"
+    ) -> Figure:
+
+        cols = self.X.columns
```

Comment:
should we assign another variable for this particular use?

Reply:
hm i think it's not that readable to uses that long expression many times